xonsh/jobs.py

```diff
@@ -219,6 +219,20 @@ def wait_for_active_job(last_task=None, backgrounded=False, _nochild=False):
         return wait_for_active_job(last_task=active_task, backgrounded=backgrounded)
 
 
+def _safe_wait_for_active_job(last_task=None, backgrounded=False):
+    """Safely call wait_for_active_job()"""
+    have_error = True
+    while have_error:
+        try:
+            rtn = wait_for_active_job(
+                last_task=last_task, backgrounded=backgrounded, return_error=True
+            )
+        except ChildProcessError as e:
+            rtn = e
+        have_error = isinstance(e, ChildProcessError)
```

Comment:
There is an issue here; `e` may not exist if an exception wasn't thrown. Should this have been indented? 

But anyway I don't think I understand things here. What is the reasoning behind `_safe_wait_for_active_job` and `wait_for_active_job` calling each other?

Reply:
Or should it be `have_error = isinstance(rtn, ChildProcessError)`

Reply:
Oh yeah, it should have been rtn. Good catch!